poly/joint.go

```diff
@@ -0,0 +1,284 @@
+package poly
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/config"
+	"io"
+)
+
+// This package provides  a dealer-less distributed verifiable secret sharing
+// using Pedersen VSS scheme as explained in "Provably Secure Distributed
+// Schnorr Signatures and a (t, n) Threshold Scheme for Implicit Certificates"
+// This file is only responsible for the setup of a shared secret among n peers.
+// The output is a global public polynomial (pubPoly) and a secret share for
+// each peers.
```

Comment:
You declare a lot of public types below in this module.  Is it actually appropriate for all of them to be public?  Are any of them unlikely to be conceivably useful to anyone outside this module?  I'm not sure yet myself, but should be considered carefully in any case, to avoid unnecessary type clutter that will make it difficult for potential users to penetrate and understand this package's API.